content/en/docs/build/dev-and-maintenance/rust/kanto-auto-deployer.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "Kanto Auto deployer"
+date: 2023-01-03T17:24:56+05:30
+weight: 2
+---
+
+**TLDR**: To deploy a container in the final Leda image, all you generally need to do is add the manifest in the [kanto-containers](https://github.com/eclipse-leda/meta-leda/tree/main/meta-leda-components/recipes-sdv/eclipse-leda/kanto-containers) directory and re-build.
+
+Kanto-CM does not provide (currently) an out-of-the box feature that allows for the automatic deployment of containers through manifest files similar to k3s' automated deployment of k8s-manifests found in the `/var/lib/rancher/k3s/server/manifests` directory.
+
+This can be worked around via a bash script for each container that runs on boot and makes sure it's deployed. Even though this approach is functional it is not very structured and would require a lot repeating code.
+
+That is why the "[Kanto Auto deployer](https://github.com/eclipse-leda/leda-utils/tree/main/src/rust/kanto-auto-deployer)" tool was developed. It directly implements the ideas in [Communicating with Кanto-CM via gRPC](../notes-on-kanto-grpc).
+
+The compiled binary takes a path to a directory containing the json manifests, parses them into Rust structures and sends gRPC requests to kanto container management to deploy these containers.
+If the container is already deployed the manifest is ignored.
+
+## Manifest structure
+
+Kanto auto deployer uses the exact same structure for its manifests as the internal representation of container state in kanto container management. For example:
+
+```json
+{
```

Comment:
This is an old example. No ports
https://github.com/eclipse-leda/meta-leda/pull/106/commits/4e43671d8cc5881b9f0e6d962f79b4412d8f61b8#diff-02e135cb4b250fa858c9b2126603111b123cd7bc45160fa470ecbdf368885fec

Reply:
Updated example manifest